src/dqrobotics/interfaces/vrep/DQ_VrepInterface.cpp

```diff
@@ -566,3 +590,531 @@ bool DQ_VrepInterface::is_video_recording()
+                      data.output_floats[3],data.output_floats[4],data.output_floats[5],
+                      data.output_floats[6],data.output_floats[7],data.output_floats[8];
+    return inertia_matrix;
+
+}
+
+
+/**
+ * @brief This method returns the inertia matrix (expressed in the shape frame) of an object on the CoppeliaSim scene.
+ * @param obj_name The name of the object where the script is attached to.
+ * @param function_name The name of the script function to call in the specified script.
+ * @param link name The name of the object from which we want to extract the inertia matrix.
+ * @returns The inertia matrix.
+ *
+ *
+ *              Example:
+ *              // This example assumes that in your CoppeliaSim there is a child script called
+ *              // "DQRoboticsApiCommandServer", where is defined the following Lua function:
+ *              //
+ *              //     function get_inertia(inInts,inFloats,inStrings,inBuffer)
+ *              //
+ *              //         if #inInts>=1 then
+ *              //            IM, matrixSFCOM=sim.getShapeInertia(inInts[1])
+ *              //
+ *              //             if inStrings[1] == 'absolute_frame' then
+ *              //                 matrix0_SF=sim.getObjectMatrix(inInts[1],-1)
+ *              //                 M = sim.multiplyMatrices(matrix0_SF,matrixSFCOM)
+ *              //                I= {{IM[1],IM[2],IM[3]},
+ *              //                     {IM[4],IM[5],IM[6]},
+ *              //                     {IM[7],IM[8],IM[9]}}
+ *              //                 R_0_COM = {{M[1],M[2],M[3]},
+ *              //                           {M[5],M[6],M[7]},
+ *              //                            {M[9],M[10],M[11]}}
+ *              //                 R_0_COM_T = transpose(R_0_COM)
+ *              //                 RIR_T = mat_mult(mat_mult(R_0_COM,I), R_0_COM_T)
+ *              //                 RIR_T_v = {RIR_T[1][1], RIR_T[1][2], RIR_T[1][3],
+ *              //                            RIR_T[2][1], RIR_T[2][2], RIR_T[2][3],
+ *              //                            RIR_T[3][1], RIR_T[3][2], RIR_T[3][3]}
+ *              //                 resultInertiaMatrix=RIR_T_v
+ *              //             else
+ *              //                 resultInertiaMatrix=IM
+ *              //             end
+ *              //             return {},resultInertiaMatrix,{},''
+ *              //         end
+ *              //     end
+ *              // In addition, it is assumed a FrankaEmikaPanda robot manipulator
+ *              // in the Coppelia scene.
+ *
+ *              DQ_VrepInterface vi;
+ *              MatrixXd inertia_matrix = vi.extract_inertia_matrix("DQRoboticsApiCommandServer","get_inertia", "Franka_link3_resp");
+ *
+ */
+MatrixXd DQ_VrepInterface::extract_inertia_matrix(const std::string&  obj_name, const std::string&  function_name,const std::string& link_name)
```

Comment:
If you use the default values I mentioned before, this overload is not necessary.